app/budget/services/budgetReducers.js

```diff
@@ -58,6 +58,15 @@ budgetApp.service('budgetReducers', function ($rootScope, $log, budgetSelectors)
 					var updatedLineItem = action.payload;
 					lineItems.list[updatedLineItem.id] = updatedLineItem;
 					return lineItems;
+				case DELETE_LINE_ITEMS:
+					action.payload.lineItemIds.forEach(function(lineItemId) {
```

Comment:
Wouldn't https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter be cleaner?

Reply:
Hmmm, unless I'm missing something, if I use filter I'd have to write two loops instead of one, and the second loop will look exactly the same.

Filter would give me a new array subset of matching values, but the values really aren't interesting here, its the values position in the original object, so that I can remove it.